website/docs/r/resource.html.markdown

```diff
@@ -35,14 +35,46 @@ EOT
 }
 ```
 
+###Using full_data
+
+```hcl
+resource "turbot_resource" "my_resource" {
+  parent   = "tmod:@turbot/turbot#/"
+  type     = "tmod:@turbot/aws#/resource/types/account"
+  full_data     = <<EOT
+{
+  "foo": "bar",
+  "title": "turbot account resource"
+}
+EOT
+}
+```
+
+###Using full_metadata
+
+```hcl
+resource "turbot_resource" "my_resource" {
+  parent   = "tmod:@turbot/turbot#/"
+  type     = "tmod:@turbot/aws#/resource/types/account"
+  full_metadata    = <<EOT
+{
+  "resource_version": "1.0.0"
+  "replication": "true"
+}
+EOT
+}
+```
+
 ## Argument Reference
 
 The following arguments are supported:
 
 - `parent` - (Required) The `id` or `aka` of the level at which the Turbot resource will be created.
 - `type` - (Required) Defines the type of the resource to be created.
-- `data` - (Required) JSON representation of the details of the resource. When parsed, it must be valid for the `type` schema.
+- `data` - (Required) JSON representation of the details of the resource, managed on merge basis. When parsed, it must be valid for the `type` schema.
```

Comment:
`data` - (Optional) JSON representation of resource properties to be managed by Terraform. These must be valid for the resource`type` schema. NOTE: If additional properties are set on the resource by other means, they are ignored by Terraform. 
`metadata` - (Optional) JSON representation of resource metadata properties to be managed by Terraform. NOTE: If additional metadata properties are set on the resource by other means, they are ignored by Terraform. 
`full_data` - (Optional) JSON representation of all resource properties to be set on the resource. These must be valid for the resource`type` schema. NOTE: If additional properties are set on the resource by other means, they are removed. 
`full_metadata` - (Optional) JSON representation of all resource metadata properties to be set on the resource. NOTE: If additional metadata properties are set on the resource by other means, they are removed.